jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/impl/HiveSqlGenerationVisitor.java

```diff
@@ -232,21 +238,17 @@ private String changeDateTrunc(SqlFunctionScalar function) throws AdapterExcepti
         return builder.toString();
     }
 
-
-    private boolean selectListRequiresCasts(SqlSelectList selectList) throws AdapterException {
-
-        // Do as if the user has all columns in select list
-        SqlStatementSelect select = (SqlStatementSelect) selectList.getParent();
-        int columnId = 0;
-        for (ColumnMetadata columnMeta : select.getFromClause().getMetadata().getColumns()) {
-            SqlColumn sqlColumn = new SqlColumn(columnId, columnMeta);
-            String typeName = ColumnAdapterNotes.deserialize(sqlColumn.getMetadata().getAdapterNotes(), sqlColumn.getMetadata().getName()).getTypeName();
-            if(typeName.equals("BINARY")  ){
-                return true;
-            }
-            ++columnId;
+    private Predicate<SqlNode> nodeRequiresCast = node -> {
```

Comment:
This again looks like redundant code, could potentially be refactored in future.

Reply:
Yes there is still code duplication in the dialects. I think we can improve this, when we refactor towards a plugin architecture.